core/sql/sqlcomp/CmpDescribe.cpp

```diff
@@ -3059,6 +3061,54 @@ short CmpDescribeSeabaseTable (
 
       outputLongLine(*space, viewtext, 0);
 
+      //display comment for VIEW
+      if (objectUID > 0)
+        {
+          if (cmpSBD.switchCompiler())
+            {
+              *CmpCommon::diags() << DgSqlCode(-CAT_UNABLE_TO_RETRIEVE_COMMENTS);
+              return -1;
+            }
+
+          ComTdbVirtObjCommentInfo objCommentInfo;
+          if (cmpSBD.getSeabaseObjectComment(objectUID, COM_VIEW_OBJECT, objCommentInfo, heap))
+            {
+              *CmpCommon::diags() << DgSqlCode(-CAT_UNABLE_TO_RETRIEVE_COMMENTS);
+              return -1;
```

Comment:
Should we switch the compiler back when going through this error path?

Reply:
Yes, we should.
I will add compiler switch-back for all these error paths.